### PR TITLE
Correct BFD diagnostic codes per RFC5880/6428

### PR DESCRIPTION
**Pull request by earies**
* (M) release/models/bfd/openconfig-bfd.yang
    - Add missing codes and adjust value mappings according to RFC

Previously, the enum variants for BFD diagnostic codes missed value (3) per
RFC5880.  This PR addresses the missing diagnostic code, correcting the
subsequent variant values and adding the reserved value (9) per RFC6428.

### CODE REVIEW

File: release/models/bfd/openconfig-bfd.yang [hunk 1]
```diff
@@ -138,48 +144,61 @@ module openconfig-bfd {
           "The BFD echo function failed - echo packets have not been
           received for the required period of time.";
       }
-      enum FORWARDING_RESET {
+      enum NEIGHBOR_SESSION_DOWN {
         value 3;
+        description
+          "The neighbor has signaled the session down.";
+      }
```
Comment (robshakir) [inline release/models/bfd/openconfig-bfd.yang:151]: Can you please add a blank like between each of these `enum` values? Not sure how we missed this originally. Thanks.

File: release/models/bfd/openconfig-bfd.yang [hunk 2]
```diff
@@ -138,48 +144,61 @@ module openconfig-bfd {
...
+      enum FORWARDING_RESET {
+        value 4;
         description
           "The forwarding plane in the local system was reset - such
           that the remote system cannot rely on the forwarding state of
           the device specifying this error code.";
       }
       enum PATH_DOWN {
-        value 4;
```
Comment (robshakir) [inline release/models/bfd/openconfig-bfd.yang:149]: Per the other change, do these values correspond to the spec? Is it better for us to just remove them all?

**Review by robshakir — CHANGES_REQUESTED** [review]

File: release/models/bfd/openconfig-bfd.yang (hunk 2 above)
Comment (earies) [reply release/models/bfd/openconfig-bfd.yang:149]: These values do now correspond to the spec but yes - we could just remove them all and state the design pattern is to never assign variant values but pay very close attention to ordering if auto assigned values do infact have side effects

https://datatracker.ietf.org/doc/html/rfc5880#section-4.1
https://datatracker.ietf.org/doc/html/rfc6428#section-5

File: release/models/bfd/openconfig-bfd.yang (hunk 1 above)
Comment (earies) [reply release/models/bfd/openconfig-bfd.yang:151]: Between each enum variant definition or between the value and description?

Looking at all other models, there are no additional lines between either ... Do you have an example in any models?

Comment (robshakir) [reply release/models/bfd/openconfig-bfd.yang:151]: Ah, apologies, yes - please ignore this. I had misparsed these as identities.

**Comment by dplore** [comment]
I am ok with the changes, but as this is technically a yang breaking change due to the enum value changes, should the major version be incremented?  

As long as we specify all or no values for the enums I think we are ok.  (ie: we should not introduce enums where some values are specified and other are not).

**Pull request closed without merging**